nixos/modules/services/home-automation/homebridge.nix

```diff
@@ -0,0 +1,549 @@
+    #     allowInsecure = true;
+    #   };
+    # }
+
+    enable = mkEnableOption "Homebridge: Homekit home automation";
+
+    user = mkOption {
+      type = str;
+      default = "homebridge";
+      description = "User to run homebridge as.";
+    };
+
+    group = mkOption {
+      type = str;
+      default = "homebridge";
+      description = "Group to run homebridge as.";
+    };
+
+    openFirewall = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for the Homebridge web interface and service.
+      '';
+    };
+
+    allowInsecure = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Allow unauthenticated requests (for easier hacking).
+        In homebridge's own installer, this is enabled by default.
+        Needs to be enabled if you want to control accessories:
+        https://github.com/homebridge/homebridge-config-ui-x/wiki/Enabling-Accessory-Control
+      '';
+    };
+
+    userStoragePath = mkOption {
+      type = str;
+      default = "/var/lib/homebridge";
+      description = ''
+        Path to store homebridge user files (needs to be writeable).
+      '';
+    };
+
+    pluginPath = mkOption {
+      type = str;
+      default = "/var/lib/homebridge/node_modules";
+      description = ''
+        Path to the plugin download directory (needs to be writeable).
+        Seems this needs to end with node_modules, as Homebridge will run npm
+        on the parent directory.
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Path to an environment-file which may contain secrets.
+      '';
+    };
+
+    settings = mkOption {
+      default = { };
+      description = ''
+        Configuration options for homebridge.
+
+        For more details, see [the homebridge documentation](https://github.com/homebridge/homebridge/wiki/Homebridge-Config-JSON-Explained).
+      '';
+      type = submodule {
+        freeformType = settingsFormat.type;
+        options = {
+          description = mkOption {
+            type = str;
+            default = "Homebridge";
+            description = "Description of the homebridge instance.";
+            readOnly = true;
+          };
+
+          bridge = mkOption {
+            description = "Homebridge Bridge options";
+            default = { };
+            type = submodule {
+              freeformType = settingsFormat.type;
+              options = {
+                name = mkOption {
+                  type = str;
+                  default = "Homebridge";
+                  description = "Name of the homebridge";
+                };
+
+                port = mkOption {
+                  type = port;
+                  default = 51826;
+                  description = "The port homebridge listens on";
+                };
+              };
+            };
+          };
```

Comment:
Same thing about inner submodule, so this can just be `bridge.name` and `bridge.port`